From 281320c6faf042464b14f3dfb9460077d8d7c878 Mon Sep 17 00:00:00 2001
From: Tanu Kaskinen <tanu.kaskinen@jollamobile.com>
Date: Wed, 5 Sep 2012 12:23:59 +0300
Subject: [PATCH 1002/1015] build: Install pulsecore headers.

This is for building out-of-tree modules. Upstream doesn't want to
support out-of-tree modules, so this patch is not upstreamable.
Mer/Nemo should try to not depend on any out-of-tree modules.
---
 Makefile.am     |   2 +-
 configure.ac    |   1 +
 pulsecore.pc.in |  12 ++++++
 src/Makefile.am | 129 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 4 files changed, 143 insertions(+), 1 deletion(-)
 create mode 100644 pulsecore.pc.in

diff --git a/Makefile.am b/Makefile.am
index b0b2553..279605c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -50,7 +50,7 @@ dist_vapi_DATA = \
 		vala/libpulse-mainloop-glib.deps vala/libpulse-mainloop-glib.vapi
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libpulse.pc libpulse-simple.pc
+pkgconfig_DATA = libpulse.pc libpulse-simple.pc pulsecore.pc
 
 if HAVE_GLIB20
 pkgconfig_DATA += \
diff --git a/configure.ac b/configure.ac
index 4854711..3ee77a0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1365,6 +1365,7 @@ man/Makefile
 libpulse.pc
 libpulse-simple.pc
 libpulse-mainloop-glib.pc
+pulsecore.pc
 doxygen/Makefile
 doxygen/doxygen.conf
 src/pulse/version.h
diff --git a/pulsecore.pc.in b/pulsecore.pc.in
new file mode 100644
index 0000000..e1111d8
--- /dev/null
+++ b/pulsecore.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=${prefix}/include
+libexecdir=@libexecdir@
+pulsetestdir=${libexecdir}/pulse-tests
+
+Name: pulsecore
+Description: Module building interface for pulseaudio
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -L${libdir}/pulseaudio -L${libdir}/pulse-@PA_MAJORMINOR@/modules -lpulsecore-@PA_MAJORMINOR@ -lpulsecommon-@PA_MAJORMINOR@ -lpulse @PTHREAD_LIBS@
+Cflags: -D_REENTRANT -D__INCLUDED_FROM_PULSE_AUDIO -DPA_DEFAULT_CONFIG_DIR=\"@PA_DEFAULT_CONFIG_DIR@\" -I${includedir}
diff --git a/src/Makefile.am b/src/Makefile.am
index 857fda3..bab318d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -741,6 +741,135 @@ libpulsecommon_@PA_MAJORMINOR@_la_LIBADD += $(DBUS_LIBS)
 endif
 
 ###################################
+#         Pulsecore headers       #
+###################################
+
+pulsecoreinclude_HEADERS = \
+		pulsecore/arpa-inet.h \
+		pulsecore/asyncmsgq.h \
+		pulsecore/asyncq.h \
+		pulsecore/atomic.h \
+		pulsecore/aupdate.h \
+		pulsecore/auth-cookie.h \
+		pulsecore/authkey.h \
+		pulsecore/avahi-wrap.h \
+		pulsecore/bitset.h \
+		pulsecore/card.h \
+		pulsecore/cli-command.h \
+		pulsecore/client.h \
+		pulsecore/cli.h \
+		pulsecore/cli-text.h \
+		pulsecore/conf-parser.h \
+		pulsecore/core-error.h \
+		pulsecore/core-format.h \
+		pulsecore/core.h \
+		pulsecore/core-rtclock.h \
+		pulsecore/core-scache.h \
+		pulsecore/core-subscribe.h \
+		pulsecore/core-util.h \
+		pulsecore/cpu-arm.h \
+		pulsecore/cpu.h \
+		pulsecore/cpu-orc.h \
+		pulsecore/cpu-x86.h \
+		pulsecore/creds.h \
+		pulsecore/database.h \
+		pulsecore/dbus-shared.h \
+		pulsecore/dbus-util.h \
+		pulsecore/device-port.h \
+		pulsecore/dynarray.h \
+		pulsecore/endianmacros.h \
+		pulsecore/esound.h \
+		pulsecore/fdsem.h \
+		pulsecore/flist.h \
+		pulsecore/g711.h \
+		pulsecore/hashmap.h \
+		pulsecore/hook-list.h \
+		pulsecore/i18n.h \
+		pulsecore/idxset.h \
+		pulsecore/iochannel.h \
+		pulsecore/ioline.h \
+		pulsecore/ipacl.h \
+		pulsecore/llist.h \
+		pulsecore/lock-autospawn.h \
+		pulsecore/log.h \
+		pulsecore/ltdl-helper.h \
+		pulsecore/macro.h \
+		pulsecore/mcalign.h \
+		pulsecore/memblock.h \
+		pulsecore/memblockq.h \
+		pulsecore/memchunk.h \
+		pulsecore/memtrap.h \
+		pulsecore/mime-type.h \
+		pulsecore/mix.h \
+		pulsecore/modargs.h \
+		pulsecore/modinfo.h \
+		pulsecore/module.h \
+		pulsecore/msgobject.h \
+		pulsecore/mutex.h \
+		pulsecore/namereg.h \
+		pulsecore/native-common.h \
+		pulsecore/object.h \
+		pulsecore/once.h \
+		pulsecore/packet.h \
+		pulsecore/parseaddr.h \
+		pulsecore/pdispatch.h \
+		pulsecore/pid.h \
+		pulsecore/pipe.h \
+		pulsecore/play-memblockq.h \
+		pulsecore/play-memchunk.h \
+		pulsecore/poll.h \
+		pulsecore/proplist-util.h \
+		pulsecore/protocol-cli.h \
+		pulsecore/protocol-dbus.h \
+		pulsecore/protocol-esound.h \
+		pulsecore/protocol-http.h \
+		pulsecore/protocol-native.h \
+		pulsecore/protocol-simple.h \
+		pulsecore/pstream.h \
+		pulsecore/pstream-util.h \
+		pulsecore/queue.h \
+		pulsecore/random.h \
+		pulsecore/ratelimit.h \
+		pulsecore/refcnt.h \
+		pulsecore/remap.h \
+		pulsecore/resampler.h \
+		pulsecore/rtkit.h \
+		pulsecore/rtpoll.h \
+		pulsecore/sample-util.h \
+		pulsecore/sconv.h \
+		pulsecore/sconv-s16be.h \
+		pulsecore/sconv-s16le.h \
+		pulsecore/semaphore.h \
+		pulsecore/shared.h \
+		pulsecore/shmasyncq.h \
+		pulsecore/shm.h \
+		pulsecore/sink.h \
+		pulsecore/sink-input.h \
+		pulsecore/sioman.h \
+		pulsecore/sndfile-util.h \
+		pulsecore/socket-client.h \
+		pulsecore/socket.h \
+		pulsecore/socket-server.h \
+		pulsecore/socket-util.h \
+		pulsecore/sound-file.h \
+		pulsecore/sound-file-stream.h \
+		pulsecore/source.h \
+		pulsecore/source-output.h \
+		pulsecore/start-child.h \
+		pulsecore/strbuf.h \
+		pulsecore/stream-util.h \
+		pulsecore/strlist.h \
+		pulsecore/svolume-orc-gen.h \
+		pulsecore/tagstruct.h \
+		pulsecore/thread.h \
+		pulsecore/thread-mq.h \
+		pulsecore/time-smoother.h \
+		pulsecore/tokenizer.h \
+		pulsecore/usergroup.h \
+		pulsecore/x11prop.h \
+		pulsecore/x11wrap.h
+
+###################################
 #         Client library          #
 ###################################
 
-- 
1.9.1

